File: Expense-Splitter/BE/src/BE.Application/DTOs/Group/GroupDto.cs
namespace BE.Application.DTOs.Group
{
    public class GroupDto
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public string? Description { get; set; }
        public string Currency { get; set; }
        public bool IsActive { get; set; }
        public string InviteCode { get; set; }
        public int MemberCount { get; set; }
        public decimal TotalExpenses { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
        public bool IsAdmin { get; set; }
        public decimal UserBalance { get; set; }
    }

    public class GroupDetailDto : GroupDto
    {
        public List<GroupMemberDto> Members { get; set; } = new();
        public GroupStatisticsDto Statistics { get; set; }
    }

    public class GroupListDto
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public int MemberCount { get; set; }
        public decimal UserBalance { get; set; }
        public DateTime LastActivity { get; set; }
        public bool IsAdmin { get; set; }
    }

    public class GroupStatisticsDto
    {
        public decimal TotalExpenses { get; set; }
        public int TotalTransactions { get; set; }
        public decimal AverageExpensePerMember { get; set; }
        public Dictionary<string, decimal> ExpensesByCategory { get; set; } = new();
        public Dictionary<Guid, UserBalanceDto> MemberBalances { get; set; } = new();
    }

    public class UserBalanceDto
    {
        public Guid UserId { get; set; }
        public string UserName { get; set; }
        public decimal Balance { get; set; }
        public decimal TotalPaid { get; set; }
        public decimal TotalOwed { get; set; }
    }
}

File: Expense-Splitter/FE/expense-splitter/src/entities/group/group.dto.ts
import { User } from "../dto";

export interface PagedList<T> {
  items: T[];
  currentPage: number;
  totalPages: number;
  pageSize: number;
  totalCount: number;
  hasPrevious: boolean;
  hasNext: boolean;
}

export interface GroupListDto {
  id: string;
  name: string;
  memberCount: number;
  userBalance: number;
  lastActivity: string;
  isAdmin: boolean;
}

export interface GroupDetailDto {
  id: string;
  name: string;
  description?: string;
  currency: string;
  inviteCode: string;
  isActive: boolean;
  createdById: string;
  createdAt: string;
  members: GroupMemberDto[];
  totalExpenses: number;
  userBalance: number;
}

export interface GroupMemberDto {
  id: string;
  userId: string;
  groupId: string;
  role: "Admin" | "Member";
  joinedAt: string;
  user: User;
  avatarUrl: string;
  balance: number;
}

export interface CreateGroupRequest {
  name: string;
  description?: string;
  currency?: string;
}

export interface UpdateGroupRequest {
  name?: string;
  description?: string;
  currency?: string;
}

export interface JoinGroupRequest {
  inviteCode: string;
}

export interface InviteMemberRequest {
  email: string;
}
